### PR TITLE
feat: add panic recovery wrapper for goroutines

### PR DESCRIPTION
**Pull request by YannickPferr**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

Adds a panic recovery wrapper that can be used when spawning goroutines. This wrapper ensures we recover from panics inside goroutines, so they don't crash our main routine.

### CODE REVIEW

File: pkg/flink/internal/utils/panic.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package utils
+
+func WithPanicRecovery(fn func()) func() {
+	return func() {
+		defer func() {
+			if r := recover(); r != nil {
+				OutputErr("Error: internal error occurred")
+			}
+		}()
+
```
Comment (luca-filipponi) [inline pkg/flink/internal/utils/panic.go:10]: unucessary space?

File: pkg/flink/internal/utils/panic_test.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+func TestPanicRecoveryWrapper(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go WithPanicRecovery(func() {
+		defer wg.Done()
+		panic("a panic in a goroutine")
+	})()
+
+	wg.Wait()
```
Comment (luca-filipponi) [inline pkg/flink/internal/utils/panic_test.go:17]: Not sure if this is a good test, if you have a bug here the test will wait indefinitely? 

If here what are you testing is the panic recovery, which advantage is this grouting giving to you?

File: pkg/flink/internal/utils/panic.go (hunk 1 above)
Comment (YannickPferr) [reply pkg/flink/internal/utils/panic.go:10]: true, will remove

File: pkg/flink/internal/utils/panic_test.go (hunk 2 above)
Comment (YannickPferr) [reply pkg/flink/internal/utils/panic_test.go:17]: Good point, but doesn't the `defer wg.Done()` ensure it's always called? 🤔 
Anyways, I think you're right, it's probably enough to just test the panic recovery behavior without the goroutine. I wanted to prove it also works with goroutines, but that might be unnecessary.

Comment (luca-filipponi) [reply pkg/flink/internal/utils/panic_test.go:17]: ooh right, you had that in a defer. So i guess that's fine, you can leave it as it

Comment (YannickPferr) [reply pkg/flink/internal/utils/panic_test.go]: No, I think you had a good point. I've refactored it a bit and now I can also reuse this wrapper in application.go (where we already had a custom panic recovery )

**Review by Guttz — APPROVED** [review]
Cool PR! Also really easy to review ✅

**Pull request merged**